cluster/photon-controller/setup-prereq.sh

```diff
@@ -0,0 +1,158 @@
+        echo "    * A disk flavor named ${PHOTON_DISK_FLAVOR}"
+        echo "It will also upload the Kube VMDK"
+        echo ""
+        echo "It creates the tenant with a resource ticket (quota) that may"
+        echo "be inappropriate for your environment. For a production"
+        echo "environment, you should configure these to match your"
+        echo "environment."
+        exit 1
+    fi
+
+    echo "Photon Target: ${PHOTON_TARGET}"
+    echo "Photon VMDK: ${PHOTON_VMDK}"
+}
+
+function set-target {
+    $PHOTON target set ${PHOTON_TARGET} > /dev/null 2>&1
+}
+
+function create-tenant {
+    local rc=0
+
+    $PHOTON tenant list | grep "\t${PHOTON_TENANT}$" > /dev/null 2>&1 || rc=$?
+    if [ $rc -eq 0 ]; then
+        echo "Tenant ${PHOTON_TENANT} already made, skipping"
+    else
+        echo "Making tenant ${PHOTON_TENANT}"
+        $PHOTON tenant create ${PHOTON_TENANT} > /dev/null 2>&1
+    fi
+    $PHOTON tenant set ${PHOTON_TENANT} > /dev/null 2>&1
+}
+
+function create-project {
+    local rc=0
+
+    $PHOTON project list | grep "\t${PHOTON_PROJECT}\t" > /dev/null 2>&1  || rc=$?
+    if [ $rc -eq 0 ]; then
+        echo "Project ${PHOTON_PROJECT} already made, skipping"
+    else
+        echo "Making project ${PHOTON_PROJECT}"
+        $PHOTON resource-ticket create --tenant ${PHOTON_TENANT} --name ${PHOTON_TENANT}-resources \
+            --limits "vm.memory 1000 GB, vm 1000 COUNT" > /dev/null 2>&1
+        $PHOTON project create --tenant ${PHOTON_TENANT} --resource-ticket ${PHOTON_TENANT}-resources \
+            --name ${PHOTON_PROJECT} --limits "vm.memory 1000 GB, vm 1000 COUNT" > /dev/null 2>&1
+    fi
+    $PHOTON project set ${PHOTON_PROJECT}
+}
+
+function create-vm-flavor {
+    local flavor=$1
+    local rc=0
+
+    $PHOTON flavor list | grep "\t${flavor}\t" > /dev/null 2>&1 || rc=$?
+    if [ $rc -eq 0 ]; then
+        echo "Flavor ${flavor} already made, skipping"
+    else
+        echo "Making VM flavor ${flavor}"
+        $PHOTON -n flavor create --name $flavor --kind "vm" --cost "vm 1 COUNT, vm.cpu 1 COUNT, vm.memory 2 GB" > /dev/null 2>&1
+    fi
+}
+
+function create-disk-flavor {
+    local rc=0
+
+    $PHOTON flavor list | grep "\t${PHOTON_DISK_FLAVOR}\t" > /dev/null 2>&1  || rc=$?
+    if [ $rc -eq 0 ]; then
+        echo "Flavor ${PHOTON_DISK_FLAVOR} already made, skipping"
+    else
+        echo "Making disk flavor ${PHOTON_DISK_FLAVOR}"
+        $PHOTON -n flavor create --name ${PHOTON_DISK_FLAVOR} --kind "ephemeral-disk" --cost "ephemeral-disk 1 COUNT" > /dev/null 2>&1
+    fi
+}
+
+function create-image {
+    local rc=0
+    local image_id=""
+
+    $PHOTON image list | grep "\tkube-image\t" > /dev/null 2>&1 || rc=$?
+    if [ $rc -eq 0 ]; then
+        echo "Image ${PHOTON_VMDK} already uploaded, skipping"
+    else
+        echo "Uploading image ${PHOTON_VMDK}"
+        $PHOTON image create $PHOTON_VMDK -n kube-image -i EAGER > /dev/null 2>&1
```

Comment:
Do we have a reason to always do EAGER replication? Do we want to provide options?